docs/adr/0005-hud-utility-version-management.md

```diff
@@ -0,0 +1,132 @@
+# ADR 0005: HUD Utility Version Management with HudHelper Factory Pattern
+
+## Status
+
+- Current Status: Proposed
+- Date of last update: 2025-09-18
+- Decision-makers: OP Engineering team
+
+## Context
+
+The Open Path HMIS Warehouse handles HUD (U.S. Department of Housing and Urban Development) data specifications that change annually or bi-annually. These specifications define data structures, validation rules, and enumerated values used throughout the system for reporting and compliance.
+
+### Current Challenges
+
+1. **Multiple HUD Utility Modules**: The codebase contains `HudUtility` (legacy/2022), `HudUtility2024`, and `HudUtility2026` modules which provide specific details for each fiscal year
+2. **Widespread Code Churn**: Each year's specification change requires updating references throughout the codebase
+3. **Maintenance Burden**: Managing the use of multiple utility classes and their usage patterns across the codebase means we need revisit each usage each year.  As the codebase grows, this leads to additional review requirements, extending the time it takes to upgrade each year
+4. **Current Year Determination**: The determination of what constitutes a current year is currently spread across the application.  This makes it difficult during critical times to find and update ALL of the places the codebase needs to be version aware
+
+### Considerations
+In identifying a solution, we would like to:
+1. **Decrease Maintenance Time**: The time it takes to upgrade each year should not grow significantly over time.  We know these changes are coming, we should prioritize making the switch as efficient as possible
+2. **Decrease Code Churn**: While we need parts of the application pinned to a specific fiscal year, much of the application can operate in the "current" fiscal year.  As long as there aren't breaking changes, we shouldn't need to update all of the usage of the utility
+3. **Long-running Job Consistency**: Jobs that span fiscal year transitions (October 1st) should consistently reference the same fiscal year information throughout
+4. **Thread Safety**: There shouldn't be a way to accidentally switch between current fiscal years
```

Comment:
```suggestion
4. **Thread Safety**: There shouldn't be a way to accidentally switch between current fiscal years. This might occur due to time-based roll over (current HUD standard changes independent of a deployment).
```